x-pack/plugins/synthetics/public/apps/synthetics/components/settings/alerting_defaults/alert_defaults_form.tsx

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiDescribedFormGroup,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiSpacer,
+} from '@elastic/eui';
+import { useDispatch, useSelector } from 'react-redux';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { i18n } from '@kbn/i18n';
+import { isEmpty, isEqual } from 'lodash';
+import { hasInvalidEmail } from './validation';
+import { DYNAMIC_SETTINGS_DEFAULTS } from '../../../../../../common/constants';
+import { DefaultEmail } from './default_email';
+import { selectDynamicSettings } from '../../../state/settings/selectors';
+import {
+  getDynamicSettingsAction,
+  setDynamicSettingsAction,
+} from '../../../state/settings/actions';
+import { DefaultConnectorField } from './connector_field';
+import { DynamicSettings } from '../../../../../../common/runtime_types';
+import { useAlertingDefaults } from './hooks/use_alerting_defaults';
+
+interface FormFields extends Omit<DynamicSettings, 'defaultEmail'> {
+  defaultEmail: Partial<DynamicSettings['defaultEmail']>;
+}
+
+export const AlertDefaultsForm = () => {
+  const dispatch = useDispatch();
+
+  const { settings, loading } = useSelector(selectDynamicSettings);
+
+  const [formFields, setFormFields] = useState<FormFields>(DYNAMIC_SETTINGS_DEFAULTS as FormFields);
+
+  const canEdit: boolean =
+    !!useKibana().services?.application?.capabilities.uptime.configureSettings || false;
+
+  const isDisabled = !canEdit;
+
+  useEffect(() => {
+    if (settings) {
+      setFormFields(settings as FormFields);
+    }
+  }, [settings]);
+
+  useEffect(() => {
+    dispatch(getDynamicSettingsAction.get());
+  }, [dispatch]);
```

Comment:
Is `dispatch` guaranteed to always be the same function? Otherwise this effect might trigger multiple times without us knowing.

Reply:
Yes, it's a redux hook return, so it will be same.